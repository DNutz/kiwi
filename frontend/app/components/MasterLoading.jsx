import React from 'react';
import Dialog from 'material-ui/Dialog';

const MasterLoading = ({activeLoading}) => (
    <Dialog
        modal={true}
        open={activeLoading}
        autoDetectWindowHeight={true}
        contentStyle={{width: '250px'}}
    >
        <div style={{display: 'flex', justifyContent: 'space-around'}}>
            <div id="wave">
                <span className="dot"></span>
                <span className="dot"></span>
                <span className="dot"></span>
            </div>
            <h1>Loading</h1>
        </div>
    </Dialog>
);

export default MasterLoading;