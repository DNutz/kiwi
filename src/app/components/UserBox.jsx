import React, { Component } from 'react';
import { connect } from 'react-redux';
import { push } from 'react-router-redux';
import Paper from 'material-ui/Paper';
import FlatButton from 'material-ui/FlatButton';
import Popover, { PopoverAnimationVertical } from 'material-ui/Popover';
import Menu from 'material-ui/Menu';
import Checkbox from 'material-ui/Checkbox';
import TextField from 'material-ui/TextField';

// - Import app components
import UserAvatar from './UserAvatar';

// - Import API
import CircleAPI from '../api/CircleAPI';

// - Import actions
import * as circleActions from '../actions/circleActions';

export class UserBox extends Component {

    /**
     * Component constructor
     * @param  {object} props is an object properties of component
     */
    constructor(props) {
        super(props);

        this.state = {
            // It will be true if user follow popover is open
            open: false,

            // The value of circle input
            circleName: '',

            // It will be true if the text field for adding group is empty
            disabledAddCircle: true
        };
    }

    handleFollowUser = (checked, cid) => {
        const { userId, user } = this.props;
        const { avatar, fullName } = user;

        if (checked) {
            this.props.addFollowingUser(cid, { avatar, userId, fullName });
        } else {
            this.props.deleteFollowingUser(cid, userId);
        }
    }

    /**
     * Handle create circle
     * 
     * @memberof UserBox
     */
    handleCreateCricle = () => {
        const { circleName } = this.state;

        if (circleName && circleName.trim() !== '') {
            this.props.createCircle(circleName);

            this.setState({
                circleName: '',
                disabledAddCircle: true
            });
        }
    }

    /**
     * Handle change group name input to the state
     * 
     * @memberof UserBox
     */
    handleChangeName = (evt) => {
        this.setState({
            circleName: evt.target.value,
            disabledAddCircle: (evt.target.value === undefined || evt.target.value.trim() === '')
        });
    }

    /**
     * Handle touch tab on follow popover
     * 
     * @memberof UserBox
     */
    handleTouchTap = (evt) => {
        // This prevents ghost click.
        event.preventDefault();

        this.setState({
            open: true,
            anchorEl: evt.currentTarget,
        });
    }

    /**
     * Handle close follow popover
     * 
     * @memberof UserBox
     */
    handleRequestClose = () => {
        this.setState({ open: false });
    }

    circleList = () => {
        const { circles, _, userBelongCircles } = this.props;

        if (circles) {
            return Object.keys(circles).map((key, index) => {
                if (key.trim() !== '-Followers') {
                    const isBelong = userBelongCircles.indexOf(key) > -1;

                    return (<Checkbox
                        key={key}
                        style={{ padding: '10px' }}
                        label={circles[key].name}
                        labelStyle={{
                            overflow: 'hidden',
                            textOverflow: 'ellipsis',
                            whiteSpace: 'nowrap',
                            width: '100%'
                        }}
                        onCheck={(evt, checked) => this.handleFollowUser(checked, key)}
                        checked={isBelong}
                    />);
                }
            });
        }
    }

    /**
     * Reneder component DOM
     * @return {react element} return the DOM which rendered by component
     */
    render() {
        return (
            <Paper style={{height: '100px', width: '100%', margin: '10', textAlign: 'center'}} zDepth={1} className='grid-cell'>
                <div style={{display: 'flex', flexDirection: 'row', alignItems: 'center', justifyContent: 'flex-start', height: '100%', position: 'relative', padding: '30px'}}>
                    <div onClick={() => this.props.goTo(`/${this.props.userId}`)} style={{ cursor: 'pointer' }}>
                        <UserAvatar fullName={this.props.fullName} fileName={this.props.avatar} size={60}/>
                    </div>
                    <div onClick={() => this.props.goTo(`/${this.props.userId}`)} style={{cursor: 'pointer', wordBreak: 'break-word', padding: '10px'}}>
                        <div style={{color: 'black', fontSize: '20px', lineHeight: '20px', marginLeft: '20px', overflow: 'hidden', textOverflow: 'ellipsis', wordBreak: 'break-word', overflow: 'hidden', textOverflow: 'ellipsis'}}>
                            {this.props.user.fullName}
                        </div>
                    </div>
                    <div>
                        <FlatButton>Add friend</FlatButton>
                        <FlatButton>Cancel request</FlatButton>
                        <FlatButton>Approve</FlatButton>
                        <FlatButton>Deny</FlatButton>
                        <FlatButton>Unfriend</FlatButton>
                    </div>
                </div>
            </Paper>
        );
    }
}


/**
 * Map dispatch to props
 * @param  {func} dispatch is the function to dispatch action to reducers
 * @param  {object} ownProps is the props belong to component
 * @return {object}          props of component
 */
const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        createCircle: (name) => dispatch(circleActions.dbAddCircle(name)),
        addFollowingUser: (cid, user) => dispatch(circleActions.dbAddFollowingUser(cid, user)),
        deleteFollowingUser: (cid, followingId) => dispatch(circleActions.dbDeleteFollowingUser(cid, followingId)),
        goTo: (url) => dispatch(push(url))

    }
}

/**
 * Map state to props
 * @param  {object} state is the obeject from redux store
 * @param  {object} ownProps is the props belong to component
 * @return {object}          props of component
 */
const mapStateToProps = (state, ownProps) => {
    const { uid } = state.authorize
    const circles = state.circle ? (state.circle.userCircles[uid] || {}) : {}
    const userBelongCircles = CircleAPI.getUserBelongCircles(circles, ownProps.userId)

    return {
        circles: circles,
        userBelongCircles: userBelongCircles || [],
        belongCirclesCount: userBelongCircles.length || 0,
        firstBelongCircle: userBelongCircles ? (circles ? circles[userBelongCircles[0]] : {}) : {},
        avatar: state.user.info && state.user.info[ownProps.userId] ? state.user.info[ownProps.userId].avatar || '' : '',
        fullName: state.user.info && state.user.info[ownProps.userId] ? state.user.info[ownProps.userId].fullName || '' : ''
    }
}

// - Connect component to redux store
export default connect(mapStateToProps, mapDispatchToProps)(UserBox)