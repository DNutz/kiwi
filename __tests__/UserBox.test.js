import React from "react";
import expect from "expect";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { UserBox } from "../src/app/components/UserBox";

Enzyme.configure({ adapter: new Adapter() });

describe("Component: UserBox", () => {
    it("renders without crashing", () => {
        const wrapper = shallow(<UserBox user={{fullName: "John Doe"}}/>);
        expect(wrapper.length).toEqual(1);
    });

    it("renders correctly", () => {
        const wrapper = shallow(<UserBox user={{fullName: "John Doe"}}/>);
        expect(wrapper).toMatchSnapshot();
    });

});