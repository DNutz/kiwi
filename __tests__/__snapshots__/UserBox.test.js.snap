// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UserBox renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserBox
    user={
      Object {
        "fullName": "John Doe",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "flex-start",
            "padding": "30px",
            "position": "relative",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <Connect(UserAvatar)
            fileName={undefined}
            fullName={undefined}
            size={60}
          />
        </div>
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "padding": "10px",
              "wordBreak": "break-word",
            }
          }
        >
          <div
            style={
              Object {
                "color": "black",
                "fontSize": "20px",
                "lineHeight": "20px",
                "marginLeft": "20px",
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "wordBreak": "break-word",
              }
            }
          >
            John Doe
          </div>
        </div>
        <div>
          <FlatButton
            disabled={false}
            fullWidth={false}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          >
            Add friend
          </FlatButton>
          <FlatButton
            disabled={false}
            fullWidth={false}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          >
            Cancel request
          </FlatButton>
          <FlatButton
            disabled={false}
            fullWidth={false}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          >
            Approve
          </FlatButton>
          <FlatButton
            disabled={false}
            fullWidth={false}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          >
            Deny
          </FlatButton>
          <FlatButton
            disabled={false}
            fullWidth={false}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          >
            Unfriend
          </FlatButton>
        </div>
      </div>,
      "circle": false,
      "className": "grid-cell",
      "rounded": true,
      "style": Object {
        "height": "100px",
        "margin": "10",
        "textAlign": "center",
        "width": "100%",
      },
      "transitionEnabled": true,
      "zDepth": 1,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <Connect(UserAvatar)
              fileName={undefined}
              fullName={undefined}
              size={60}
            />
          </div>,
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "padding": "10px",
                "wordBreak": "break-word",
              }
            }
          >
            <div
              style={
                Object {
                  "color": "black",
                  "fontSize": "20px",
                  "lineHeight": "20px",
                  "marginLeft": "20px",
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "wordBreak": "break-word",
                }
              }
            >
              John Doe
            </div>
          </div>,
          <div>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Add friend
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Cancel request
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Approve
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Deny
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Unfriend
            </FlatButton>
          </div>,
        ],
        "style": Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "flex-start",
          "padding": "30px",
          "position": "relative",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(UserAvatar)
              fileName={undefined}
              fullName={undefined}
              size={60}
            />,
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fileName": undefined,
              "fullName": undefined,
              "size": 60,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              style={
                Object {
                  "color": "black",
                  "fontSize": "20px",
                  "lineHeight": "20px",
                  "marginLeft": "20px",
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "wordBreak": "break-word",
                }
              }
            >
              John Doe
            </div>,
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
              "padding": "10px",
              "wordBreak": "break-word",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "John Doe",
              "style": Object {
                "color": "black",
                "fontSize": "20px",
                "lineHeight": "20px",
                "marginLeft": "20px",
                "overflow": "hidden",
                "textOverflow": "ellipsis",
                "wordBreak": "break-word",
              },
            },
            "ref": null,
            "rendered": "John Doe",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Add friend
              </FlatButton>,
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Cancel request
              </FlatButton>,
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Approve
              </FlatButton>,
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Deny
              </FlatButton>,
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Unfriend
              </FlatButton>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add friend",
                "disabled": false,
                "fullWidth": false,
                "labelPosition": "after",
                "labelStyle": Object {},
                "onKeyboardFocus": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "onTouchStart": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": "Add friend",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel request",
                "disabled": false,
                "fullWidth": false,
                "labelPosition": "after",
                "labelStyle": Object {},
                "onKeyboardFocus": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "onTouchStart": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": "Cancel request",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Approve",
                "disabled": false,
                "fullWidth": false,
                "labelPosition": "after",
                "labelStyle": Object {},
                "onKeyboardFocus": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "onTouchStart": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": "Approve",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Deny",
                "disabled": false,
                "fullWidth": false,
                "labelPosition": "after",
                "labelStyle": Object {},
                "onKeyboardFocus": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "onTouchStart": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": "Deny",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Unfriend",
                "disabled": false,
                "fullWidth": false,
                "labelPosition": "after",
                "labelStyle": Object {},
                "onKeyboardFocus": [Function],
                "onMouseEnter": [Function],
                "onMouseLeave": [Function],
                "onTouchStart": [Function],
                "primary": false,
                "secondary": false,
              },
              "ref": null,
              "rendered": "Unfriend",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "height": "100%",
              "justifyContent": "flex-start",
              "padding": "30px",
              "position": "relative",
            }
          }
        >
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <Connect(UserAvatar)
              fileName={undefined}
              fullName={undefined}
              size={60}
            />
          </div>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "padding": "10px",
                "wordBreak": "break-word",
              }
            }
          >
            <div
              style={
                Object {
                  "color": "black",
                  "fontSize": "20px",
                  "lineHeight": "20px",
                  "marginLeft": "20px",
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "wordBreak": "break-word",
                }
              }
            >
              John Doe
            </div>
          </div>
          <div>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Add friend
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Cancel request
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Approve
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Deny
            </FlatButton>
            <FlatButton
              disabled={false}
              fullWidth={false}
              labelPosition="after"
              labelStyle={Object {}}
              onKeyboardFocus={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onTouchStart={[Function]}
              primary={false}
              secondary={false}
            >
              Unfriend
            </FlatButton>
          </div>
        </div>,
        "circle": false,
        "className": "grid-cell",
        "rounded": true,
        "style": Object {
          "height": "100px",
          "margin": "10",
          "textAlign": "center",
          "width": "100%",
        },
        "transitionEnabled": true,
        "zDepth": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <Connect(UserAvatar)
                fileName={undefined}
                fullName={undefined}
                size={60}
              />
            </div>,
            <div
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                  "padding": "10px",
                  "wordBreak": "break-word",
                }
              }
            >
              <div
                style={
                  Object {
                    "color": "black",
                    "fontSize": "20px",
                    "lineHeight": "20px",
                    "marginLeft": "20px",
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "wordBreak": "break-word",
                  }
                }
              >
                John Doe
              </div>
            </div>,
            <div>
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Add friend
              </FlatButton>
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Cancel request
              </FlatButton>
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Approve
              </FlatButton>
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Deny
              </FlatButton>
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Unfriend
              </FlatButton>
            </div>,
          ],
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "flex-start",
            "padding": "30px",
            "position": "relative",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(UserAvatar)
                fileName={undefined}
                fullName={undefined}
                size={60}
              />,
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "fileName": undefined,
                "fullName": undefined,
                "size": 60,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                style={
                  Object {
                    "color": "black",
                    "fontSize": "20px",
                    "lineHeight": "20px",
                    "marginLeft": "20px",
                    "overflow": "hidden",
                    "textOverflow": "ellipsis",
                    "wordBreak": "break-word",
                  }
                }
              >
                John Doe
              </div>,
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
                "padding": "10px",
                "wordBreak": "break-word",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "John Doe",
                "style": Object {
                  "color": "black",
                  "fontSize": "20px",
                  "lineHeight": "20px",
                  "marginLeft": "20px",
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "wordBreak": "break-word",
                },
              },
              "ref": null,
              "rendered": "John Doe",
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  labelPosition="after"
                  labelStyle={Object {}}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={false}
                  secondary={false}
                >
                  Add friend
                </FlatButton>,
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  labelPosition="after"
                  labelStyle={Object {}}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={false}
                  secondary={false}
                >
                  Cancel request
                </FlatButton>,
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  labelPosition="after"
                  labelStyle={Object {}}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={false}
                  secondary={false}
                >
                  Approve
                </FlatButton>,
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  labelPosition="after"
                  labelStyle={Object {}}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={false}
                  secondary={false}
                >
                  Deny
                </FlatButton>,
                <FlatButton
                  disabled={false}
                  fullWidth={false}
                  labelPosition="after"
                  labelStyle={Object {}}
                  onKeyboardFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onTouchStart={[Function]}
                  primary={false}
                  secondary={false}
                >
                  Unfriend
                </FlatButton>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Add friend",
                  "disabled": false,
                  "fullWidth": false,
                  "labelPosition": "after",
                  "labelStyle": Object {},
                  "onKeyboardFocus": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onTouchStart": [Function],
                  "primary": false,
                  "secondary": false,
                },
                "ref": null,
                "rendered": "Add friend",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Cancel request",
                  "disabled": false,
                  "fullWidth": false,
                  "labelPosition": "after",
                  "labelStyle": Object {},
                  "onKeyboardFocus": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onTouchStart": [Function],
                  "primary": false,
                  "secondary": false,
                },
                "ref": null,
                "rendered": "Cancel request",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Approve",
                  "disabled": false,
                  "fullWidth": false,
                  "labelPosition": "after",
                  "labelStyle": Object {},
                  "onKeyboardFocus": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onTouchStart": [Function],
                  "primary": false,
                  "secondary": false,
                },
                "ref": null,
                "rendered": "Approve",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Deny",
                  "disabled": false,
                  "fullWidth": false,
                  "labelPosition": "after",
                  "labelStyle": Object {},
                  "onKeyboardFocus": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onTouchStart": [Function],
                  "primary": false,
                  "secondary": false,
                },
                "ref": null,
                "rendered": "Deny",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Unfriend",
                  "disabled": false,
                  "fullWidth": false,
                  "labelPosition": "after",
                  "labelStyle": Object {},
                  "onKeyboardFocus": [Function],
                  "onMouseEnter": [Function],
                  "onMouseLeave": [Function],
                  "onTouchStart": [Function],
                  "primary": false,
                  "secondary": false,
                },
                "ref": null,
                "rendered": "Unfriend",
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
